src/editors/components/EditorFooter/index.test.jsx

```diff
@@ -1,104 +1,96 @@
+import { RequestKeys } from '../../data/constants/requests';
+import { saveTextBlock, navigateCallback } from '../../hooks';
 
-const locationTemp = window.location;
-beforeAll(() => {
-  delete window.location;
-  window.location = {
-    assign: jest.fn(),
-  };
-});
-afterAll(() => {
-  window.location = locationTemp;
-});
+jest.mock('../../data/redux', () => ({
+  thunkActions: {
+    app: {
+      saveBlock: jest.fn().mockName('thunkActions.app.saveBlock'),
+    },
+  },
+  selectors: {
+    app: {
+      isInitialized: jest.fn(state => ({ isInitialized: state })),
+      studioEndpointUrl: jest.fn(state => ({ studioEndpointUrl: state })),
+    },
+    requests: {
+      // eslint-disable-next-line no-unused-vars
+      isFailed: jest.fn((state, requestkey) => ({ isFailed: state })),
+    },
+  },
+}));
 
-jest.mock('./data/api', () => {
-  const originalModule = jest.requireActual('./data/api');
-  // Mock the default export and named export saveBlock
+jest.mock('.', () => {
+  // Require the original module to not be mocked...
+  const originalModule = jest.requireActual('./index');
   return {
-    __esModule: true,
+    __esModule: true, // Use it when dealing with esModules
     ...originalModule,
-    saveBlock: jest.fn(() => {}),
-  };
-});
-
-jest.spyOn(React, 'useRef').mockReturnValue({
-  current: {
-    getContent: () => '',
-  },
-});
-
-test('Rendering: loaded', () => {
-  const context = {
-    unitUrlLoading: ActionStates.FINISHED,
-  };
-  render(
-    <EditorPageContext.Provider value={context}>
-      <EditorFooter />
-    </EditorPageContext.Provider>,
-  );
-  expect(screen.getByText('Cancel')).toBeTruthy();
-  expect(screen.getByText('Add To Course')).toBeTruthy();
-});
-
-test('Rendering: loading url', () => {
-  const context = {
-    unitUrlLoading: ActionStates.NOT_BEGUN,
+    handleCancelClicked: jest.fn().mockName('handleCancelClicked'),
+    handleSaveClicked: jest.fn().mockName('handleSaveClicked'),
   };
-  render(
-    <EditorPageContext.Provider value={context}>
-      <EditorFooter />
-    </EditorPageContext.Provider>,
-  );
-  expect(screen.getByText('Cancel')).toBeTruthy();
-  expect(screen.getAllByRole('button', { 'aria-label': 'Save' })).toBeTruthy();
-  expect(screen.queryByText('Add To Course')).toBeNull();
 });
 
-test('Navigation: Cancel', () => {
-  const context = {
-    unitUrlLoading: ActionStates.FINISHED,
-    unitUrl: {
-      data: {
-        ancestors:
-        [
-          { id: 'fakeblockid' },
-        ],
-      },
-    },
-    studioEndpointUrl: 'Testurl',
-  };
-  render(
-    <EditorPageContext.Provider value={context}>
-      <EditorFooter />
-    </EditorPageContext.Provider>,
-  );
-  expect(screen.getByText('Cancel')).toBeTruthy();
-  userEvent.click(screen.getByText('Cancel'));
-  expect(window.location.assign).toHaveBeenCalled();
-});
+jest.mock('../../hooks', () => ({
+  saveTextBlock: jest.fn(),
+  navigateCallback: jest.fn(),
```

Comment:
must also include nullMethod in this mock.
```suggestion
  navigateCallback: jest.fn(),
  nullMethod: jest.fn().mockName('nullMethod'),
```